src/Http/WebUtilities/src/MultipartSectionPipeReader.cs

```diff
@@ -0,0 +1,600 @@
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.IO.Pipelines;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.WebUtilities
+{
+    public class MultipartSectionPipeReader
```

Comment:
This should derive from pipe reader and only expose that API.

Reply:
Note to self: I'm going to have to come back and read this file more closely, I can't quite wrap my head around it yet.

Reply:
I am not sure that it should be derived from PipeReader though - this way I can create only the functions that I need to support from within the type - like DrainAsync and ReadToEnd. I don't really see the benefit of deriving from a PipeReader, and limiting my self in that sense. I could create DrainAsync, ReadAsync and ReadToEnd as Extension Methods - but imo this depends on whether it will be useful in other contexts.

Reply:
The point of deriving from PipeReader is that PipeReader is an API better suited for general consumption, not something specialized. Ideally MultipartSectionPipeReader would be internal and people would only see the PipeReader.